src/main/java/com/redhat/devtools/lsp4ij/client/features/LSPFormattingFeature.java

```diff
@@ -219,15 +219,16 @@ public boolean isFormatOnStatementTerminator(@NotNull PsiFile file) {
     }
 
     /**
-     * The specific statement terminator characters that should trigger client-side on-type formatting.
+     * The specific statement terminator characters that should trigger client-side on-type formatting. Defaults to
+     * {@link LSPClientFeatures#getStatementTerminatorCharacters(PsiFile)}.
      *
      * @param file the file
      * @return the statement terminator characters that should trigger on-type formatting
      */
-    @Nullable
+    @NotNull

```

Comment:
This is now always non-null as it will default to the language grammar-specific statement terminator characters if unspecified, and those are an empty string at worst.